array.c

```diff
@@ -7894,6 +7894,45 @@ rb_ary_sum(int argc, VALUE *argv, VALUE ary)
     return v;
 }
 
+/*
+ * call-seq:
+ *   array.mean() -> object
+ *   array.mean() {|element| ... } -> object
+ *
+ *  When no block is given, returns the object equivalent to:
+ *    sum = 0
+ *    array.each {|element| sum += element }
+ *    sum / size
+ *  For example, <tt>[e1, e2, e3].mean</tt> returns <tt>(e1 + e2 + e3) / 3</tt>.
+ *
+ *  Examples:
+ *    a = [0, 1, 2, 3]
+ *    a.mean # => 1.5
+ *
+ *  When a block is given, it is called with each element
+ *  and the block's return value (instead of the element itself) is used as the addend:
+ *    a = [-3, -2, -1]
+ *    s = a.mean { |element| element.abs }
+ *    s # => 2
+ *
+ */
+
+static VALUE
+rb_ary_mean(int argc, VALUE *argv, VALUE ary)
+{
+    long len = RARRAY_LEN(ary);
+
+    if (len == 0)
+        return DBL2NUM(nan(""));
+
+    VALUE sum, average;
```

Comment:
Just a suggestion: shouldn't average be named mean, thus making the function return mean instead of average? This change would be in line with the change in the function name as suggested in the issue tracker here: https://bugs.ruby-lang.org/issues/18057#note-8